tests/merch.test.ts

```diff
@@ -88,3 +92,277 @@ describe('collection editing', () => {
+
+    const merchItemResponse = await ControllerFactory.merchStore(conn).getOneMerchItem({ uuid: item.uuid }, admin);
+
+    expect(merchItemResponse.item.options).toEqual(
+      expect.arrayContaining(options.map((option) => option.getPublicMerchItemOption())),
+    );
+  });
+
+  test('fails when variants are updated to disabled but multiple options still remain', async () => {
+    const conn = await DatabaseConnection.get();
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const options = MerchFactory.createOptions(3);
+    const [item] = MerchFactory.itemsWith({
+      hasVariantsEnabled: true,
+      options,
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+    });
+
+    await new PortalState()
+      .createUsers([admin])
+      .createMerch([collection])
+      .write();
+
+    await expect(ControllerFactory.merchStore(conn).editMerchItem(
+      { uuid: item.uuid },
+      { merchandise: { hasVariantsEnabled: false } },
+      admin,
+    )).rejects.toThrow('Merch items with variants disabled cannot have multiple options');
+  });
+
+  test('fails when updated options have multiple types', async () => {
+    const conn = await DatabaseConnection.get();
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const options = MerchFactory.createOptions(3);
+    const [item] = MerchFactory.itemsWith({
+      hasVariantsEnabled: true,
+      options,
+    });
+    const [collection] = MerchFactory.collectionsWith({
+      items: [item],
+    });
+
+    await new PortalState()
+      .createUsers([admin])
+      .createMerch([collection])
+      .write();
+
+    // change only one option's type to a different one
+    options[0].metadata.type = 'new type';
+
+    await expect(ControllerFactory.merchStore(conn).editMerchItem(
+      { uuid: item.uuid },
+      { merchandise: { options } },
+      admin,
+    )).rejects.toThrow('Merch items cannot have multiple option types');
+  });
+});
+
+describe('merch item options', () => {
+  test('can be added to an item with variants enabled and with same option type', async () => {
+    const conn = await DatabaseConnection.get();
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [metadata, newMetadata] = MerchFactory.createOptionMetadata(2);
+    const [option, newOption] = MerchFactory.optionsWith({ metadata }, { metadata: newMetadata });
+    const [merchItem] = MerchFactory.itemsWith({
+      hasVariantsEnabled: true,
+      options: [option],
+    });
+    const [merchCollection] = MerchFactory.collectionsWith({
+      items: [merchItem],
+    });
+
+    await new PortalState()
+      .createUsers([admin])
+      .createMerch([merchCollection])
+      .write();
+
+    await ControllerFactory
+      .merchStore(conn)
+      .createMerchItemOption({ uuid: merchItem.uuid }, { option: newOption }, admin);
+
+    const merchItemResponse = await ControllerFactory
```

Comment:
Nit: we can create a controller once within each test instead of using `ControllerFactory` every time.